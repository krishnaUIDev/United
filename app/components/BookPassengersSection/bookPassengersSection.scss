@import "../../styles/global/_variables";

.dropdownContainer {
  width: 100%;
  margin-bottom: 12px;
}

.section {
  margin-bottom: 16px;
  margin-top: 0;
  height: 24px;
  @media screen and (min-width: $break-point-d) {
    width: 100%;
  }
}

.sectionTitle {
  color: $black-60;
  font-size: 16px;
  letter-spacing: 0;
  line-height: 26px;
  &:focus, &:active {
    outline: 1px dotted $modern-gold;
  }
  @media screen and (min-width: $break-point-d) {
    font-size: 18px;
  }
}

.sectionDetail {
  font-size: 12px;
}

.incrementerDiv {
  margin-top: -26px;
  position: absolute;
  right: 24px;
  @media screen and (min-width: $break-point-d) {
    padding-left: 160px;
    right: auto;
  }
}

input.qty {
  width: 40px;
  height: 25px;
  text-align: center;
  font-size: 18px;
  color: $link-blue;
  letter-spacing: -0.18px;
  line-height: 1;
  border: none;
  padding: 0;
  &:focus {
    outline: 1px dotted $modern-gold;
  }
}

.qtyplus, .qtyminus {
  font-size: 18px;
  color: $link-blue;
  &:disabled {
    background: white;
    color: $black-60;
  }
}


/* Remove up/down arrows that automatically appear with input="number" */
input[type="number"]::-webkit-outer-spin-button, input[type="number"]::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
}

input[type="number"] {
    -moz-appearance: textfield;
}
