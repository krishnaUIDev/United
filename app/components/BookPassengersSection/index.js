/*
 * BookPassengers
 */

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Control } from 'react-redux-form/lib/immutable';

import styles from './bookPassengersSection.scss';

class BookPassengersSection extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const currentSelected = parseInt(this.props.index, 10) + 1;
    const ariaLabel = `${this.props.currentAmount} ${this.props.sectionTitle}, ${this.props.sectionDetail}, ${currentSelected} of ${this.props.numberOfTravelersTotal} passenger types`;

    const minValLimit = (this.props.sectionTitle === this.props.minValueIsOne) ? 1 : 0;
    const disableMinus = (this.props.disableIndividualSections) ? (this.props.currentAmount === minValLimit) : this.props.minusBtnDisabled;
    const disablePlus = (this.props.disableIndividualSections) ? (this.props.currentAmount === this.props.individualMaxLimit) : this.props.plusBtnDisabled;

    return (
      <div
        className={styles.section}
        id={this.props.id}
        aria-label="Passengers"
      >
        <span
          className={styles.sectionTitle}
          id={`${this.props.id} title`}
        >{this.props.sectionTitle}
          <span className={styles.sectionDetail}> {this.props.sectionDetail}</span>
        </span>
        <p className={styles.incrementerDiv}>
          <button
            id={`${this.props.id} minusBtn`}
            onClick={(event) => this.props.onQuantityClick(event, this.props.id, 'minus', this.props.modelName)}
            aria-label="minus"
            className={classNames(styles.qtyminus, `${this.props.id}qtyminus`)}
            disabled={disableMinus}
            tabIndex={this.props.decrementTabIndex}
            role="button"
            type="button"
          >-</button>
          <Control.input
            type="number"
            name="quantity"
            min={this.props.min}
            max={this.props.maxLimit}
            className={classNames(styles.qty, `${this.props.sectionDetail}txtInput`)}
            id={this.props.sectionTitle}
            onKeyUp={(event) => this.props.onTextInput(event, this.props.id, this.props.modelName)}
            onFocus={(event) => this.props.onQuantityInputFocus(event)}
            onClick={(event) => this.props.onQuantityInputFocus(event)}
            aria-label={ariaLabel}
            autoComplete="off"
            model={this.props.modelName}
            tabIndex="0"
            pattern="[0-9]*"
          />
          <button
            id={`${this.props.id} plusBtn`}
            onClick={(event) => this.props.onQuantityClick(event, this.props.id, 'plus', this.props.modelName, this.props.maxLimit)}
            aria-label="plus"
            disabled={disablePlus}
            className={classNames(styles.qtyplus, `${this.props.sectionDetail}qtyplus`)}
            role="button"
            type="button"
            tabIndex="0"
          >+</button>
        </p>
      </div>
    );
  }
}

BookPassengersSection.propTypes = {
  min: PropTypes.number,
  maxLimit: PropTypes.number,
  sectionTitle: PropTypes.string,
  sectionDetail: PropTypes.string,
  id: PropTypes.string,
  onTextInput: PropTypes.func,
  onQuantityClick: PropTypes.func,
  minusBtnDisabled: PropTypes.bool,
  plusBtnDisabled: PropTypes.bool,
  currentAmount: PropTypes.number,
  onQuantityInputFocus: PropTypes.func,
  decrementTabIndex: PropTypes.number,
  index: PropTypes.string,
  numberOfTravelersTotal: PropTypes.string,
  modelName: PropTypes.string,
  disableIndividualSections: PropTypes.string,
  individualMaxLimit: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  minValueIsOne: PropTypes.string,
};

export default BookPassengersSection;
