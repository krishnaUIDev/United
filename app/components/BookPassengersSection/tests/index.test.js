import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';

import BookPassengersSection from '../index';

describe('<BookPassengersSection />', () => {
  const onClick = spy();
  const onInputClickSpy = spy();
  const onKeyUpSpy = spy();
  const renderedComponent = shallow(
    <BookPassengersSection
      id="Adults (18-64)"
      sectionTitle="Adults"
      sectionDetail="(18-64)"
      maxLimit={9}
      onTextInput={onKeyUpSpy}
      onQuantityClick={onClick}
      minusBtnDisabled={false}
      plusBtnDisabled={false}
      currentAmount={3}
      onQuantityInputFocus={onInputClickSpy}
    />
  );
  const wrapper = renderedComponent.instance();
  it('render BookPassengersSection Component', () => {
    expect(renderedComponent).toMatchSnapshot();
  });
  it('render component prop sectionTitle', () => {
    expect(wrapper.props.sectionTitle).toEqual('Adults');
  });
  it('render component prop sectionDetail', () => {
    expect(wrapper.props.sectionDetail).toEqual('(18-64)');
  });
  it('should render divs', () => {
    expect(renderedComponent.find('div').length).toEqual(1);
  });
  it('should render buttons', () => {
    expect(renderedComponent.find('button').length).toEqual(2);
  });
  it('should render p', () => {
    expect(renderedComponent.find('p').length).toEqual(1);
  });
  it('should render Control.input', () => {
    expect(renderedComponent.find('#Adults').length).toEqual(1);
  });
  it('should handle minus button click', () => {
    const btn = renderedComponent.find('.(18-64)qtyminus');
    btn.simulate('click');
    expect(onClick.called).toEqual(true);
  });
  it('should handle plus button click', () => {
    const plusBtn = renderedComponent.find('.(18-64)qtyplus');
    plusBtn.simulate('click');
    expect(onClick.called).toEqual(true);
  });
  it('should render input onFocus', () => {
    renderedComponent.find('#Adults').simulate('focus');
    expect(onInputClickSpy.called).toEqual(true);
  });
  it('should render input onKeyUp', () => {
    renderedComponent.find('#Adults').simulate('keyup');
    expect(onKeyUpSpy.called).toEqual(true);
  });
});
