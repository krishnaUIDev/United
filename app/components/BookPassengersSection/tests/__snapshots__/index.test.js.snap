// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookPassengersSection /> render BookPassengersSection Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    aria-label="Passengers"
    className={undefined}
    id="Adults (18-64)"
>
    <span
        className={undefined}
        id="Adults (18-64) title"
    >
        Adults
        <span
            className={undefined}
        >
             
            (18-64)
        </span>
    </span>
    <p
        className={undefined}
    >
        <button
            aria-label="minus"
            className="Adults (18-64)qtyminus"
            disabled={false}
            id="Adults (18-64) minusBtn"
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
            type="button"
        >
            -
        </button>
        <DefaultConnectedControlInput
            aria-label="3 Adults, (18-64), NaN of undefined passenger types"
            autoComplete="off"
            className="(18-64)txtInput"
            id="Adults"
            max={9}
            min={undefined}
            model={undefined}
            name="quantity"
            onClick={[Function]}
            onFocus={[Function]}
            onKeyUp={[Function]}
            pattern="[0-9]*"
            tabIndex="0"
            type="number"
        />
        <button
            aria-label="plus"
            className="(18-64)qtyplus"
            disabled={false}
            id="Adults (18-64) plusBtn"
            onClick={[Function]}
            role="button"
            tabIndex="0"
            type="button"
        >
            +
        </button>
    </p>
</div>,
  "nodes": Array [
    <div
      aria-label="Passengers"
      className={undefined}
      id="Adults (18-64)"
>
      <span
            className={undefined}
            id="Adults (18-64) title"
      >
            Adults
            <span
                  className={undefined}
            >
                   
                  (18-64)
            </span>
      </span>
      <p
            className={undefined}
      >
            <button
                  aria-label="minus"
                  className="Adults (18-64)qtyminus"
                  disabled={false}
                  id="Adults (18-64) minusBtn"
                  onClick={[Function]}
                  role="button"
                  tabIndex={undefined}
                  type="button"
            >
                  -
            </button>
            <DefaultConnectedControlInput
                  aria-label="3 Adults, (18-64), NaN of undefined passenger types"
                  autoComplete="off"
                  className="(18-64)txtInput"
                  id="Adults"
                  max={9}
                  min={undefined}
                  model={undefined}
                  name="quantity"
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyUp={[Function]}
                  pattern="[0-9]*"
                  tabIndex="0"
                  type="number"
            />
            <button
                  aria-label="plus"
                  className="(18-64)qtyplus"
                  disabled={false}
                  id="Adults (18-64) plusBtn"
                  onClick={[Function]}
                  role="button"
                  tabIndex="0"
                  type="button"
            >
                  +
            </button>
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <BookPassengersSection
        currentAmount={3}
        id="Adults (18-64)"
        maxLimit={9}
        minusBtnDisabled={false}
        onQuantityClick={[Function]}
        onQuantityInputFocus={[Function]}
        onTextInput={[Function]}
        plusBtnDisabled={false}
        sectionDetail="(18-64)"
        sectionTitle="Adults"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": BookPassengersSection {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "currentAmount": 3,
          "id": "Adults (18-64)",
          "maxLimit": 9,
          "minusBtnDisabled": false,
          "onQuantityClick": [Function],
          "onQuantityInputFocus": [Function],
          "onTextInput": [Function],
          "plusBtnDisabled": false,
          "sectionDetail": "(18-64)",
          "sectionTitle": "Adults",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          aria-label="Passengers"
          className={undefined}
          id="Adults (18-64)"
>
          <span
                    className={undefined}
                    id="Adults (18-64) title"
          >
                    Adults
                    <span
                              className={undefined}
                    >
                               
                              (18-64)
                    </span>
          </span>
          <p
                    className={undefined}
          >
                    <button
                              aria-label="minus"
                              className="Adults (18-64)qtyminus"
                              disabled={false}
                              id="Adults (18-64) minusBtn"
                              onClick={[Function]}
                              role="button"
                              tabIndex={undefined}
                              type="button"
                    >
                              -
                    </button>
                    <DefaultConnectedControlInput
                              aria-label="3 Adults, (18-64), NaN of undefined passenger types"
                              autoComplete="off"
                              className="(18-64)txtInput"
                              id="Adults"
                              max={9}
                              min={undefined}
                              model={undefined}
                              name="quantity"
                              onClick={[Function]}
                              onFocus={[Function]}
                              onKeyUp={[Function]}
                              pattern="[0-9]*"
                              tabIndex="0"
                              type="number"
                    />
                    <button
                              aria-label="plus"
                              className="(18-64)qtyplus"
                              disabled={false}
                              id="Adults (18-64) plusBtn"
                              onClick={[Function]}
                              role="button"
                              tabIndex="0"
                              type="button"
                    >
                              +
                    </button>
          </p>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          aria-label="Passengers"
          className={undefined}
          id="Adults (18-64)"
>
          <span
                    className={undefined}
                    id="Adults (18-64) title"
          >
                    Adults
                    <span
                              className={undefined}
                    >
                               
                              (18-64)
                    </span>
          </span>
          <p
                    className={undefined}
          >
                    <button
                              aria-label="minus"
                              className="Adults (18-64)qtyminus"
                              disabled={false}
                              id="Adults (18-64) minusBtn"
                              onClick={[Function]}
                              role="button"
                              tabIndex={undefined}
                              type="button"
                    >
                              -
                    </button>
                    <DefaultConnectedControlInput
                              aria-label="3 Adults, (18-64), NaN of undefined passenger types"
                              autoComplete="off"
                              className="(18-64)txtInput"
                              id="Adults"
                              max={9}
                              min={undefined}
                              model={undefined}
                              name="quantity"
                              onClick={[Function]}
                              onFocus={[Function]}
                              onKeyUp={[Function]}
                              pattern="[0-9]*"
                              tabIndex="0"
                              type="number"
                    />
                    <button
                              aria-label="plus"
                              className="(18-64)qtyplus"
                              disabled={false}
                              id="Adults (18-64) plusBtn"
                              onClick={[Function]}
                              role="button"
                              tabIndex="0"
                              type="button"
                    >
                              +
                    </button>
          </p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BookPassengersSection
    currentAmount={3}
    id="Adults (18-64)"
    maxLimit={9}
    minusBtnDisabled={false}
    onQuantityClick={[Function]}
    onQuantityInputFocus={[Function]}
    onTextInput={[Function]}
    plusBtnDisabled={false}
    sectionDetail="(18-64)"
    sectionTitle="Adults"
/>,
}
`;
